projects/uvis_gs_fail/data_process.py

```diff
@@ -0,0 +1,238 @@
+# Imports:
+import os
+import sys
+from multiprocessing import Pool
+import numpy as np
+import pandas as pd
+from astropy.io import fits
+import tqdm
+from skimage.transform import resize, rotate
+
+
+
+def log_image_process(fpath, orig_data=False):
+    '''
+    Opens a flt.fits file and logarithmically processes the data according to 
+    the following method:
+    - opens the flt file, extracts SCI array, and creates an array of image 
+    data.
+    - Sets all values in the image array that are less than one to be equal to 
+    1.
+    - Scales the image data logarithmically.
+    - Uses min/max scaling to scale the pixel values to be between 0 and 1. 
+    - Resizes the image array to the dimensions (256, 256).
+
+    Parameters:
+    -----------
+    fpath: string
+    The string that identifies the directory that the image is located in.
+
+    Returns: numpy array of floats
+    --------
+    log_data: array of floats
+    Array of logarithmically scaled image data.
+
+    rootname: string
+    A String containing the rootname of the image that was processed.
+    '''
+
+     #Unpacking the image data: 
+        # get the boolean value of the subarray
+    subarray_value = fits.getval(fpath, 'subarray')
+    
+    # If the subarray is false then it is full chip, so we get the 
+    # two data arrays, minus the 6 dead rows of pixels
+    if subarray_value == False:
+        data1 = fits.getdata(fpath,'sci', 1)[:-3]
+        data2 = fits.getdata(fpath, 'sci', 2)[3:]
+        
+        image_data = np.vstack((data1, data2))
+        
+    else:
+        
+        image_data = fits.getdata(fpath, 'sci', 1)
+    
+        # Clip low values 
```

Comment:
Untab clip low values, log scale data, resize log scaled image, and min/max scale image array